### ISSUE TITLE
[v0.21.0] Unable to pass complex env to pack build

### ISSUE THREAD

**Issue by arjun024** [buildpacks/pack#1288]
### Summary
With v0.21, `pack build` fails when passed a complex build time env variable
like `--env BP_GO_GENERATE_FLAGS='-run "^//go:generate sleep"'`

---

##### Steps

1. Clone and change directory to https://github.com/paketo-community/go-generate/blob/v0.0.5/integration/testdata/args_and_flags/main.go

2.  v0.20.0 (Success)

```
$ pack --version
0.20.0+git-66a4f32.build-2668

$ pack build goimage --builder paketobuildpacks/builder:full \
--env BP_GO_GENERATE_FLAGS='-run "^//go:generate sleep"'
...
...
Successfully built image goimage
```

3. v0.21.0 (Failure)
```
$ /tmp/pack21 --version
0.21.0+git-a9c294a.build-2814

$ /tmp/pack21 build goimage21 --builder paketobuildpacks/builder:full \
--env BP_GO_GENERATE_FLAGS='-run "^//go:generate sleep"'
ERROR: failed to parse env line 'BP_GO_GENERATE_FLAGS=-run "^//go:generate sleep"': invalid syntax between 0 and 48
```

### More Info

See Github Action failures associated with the 2 following Pull Requests that updates pack cli on integration test suite run by Paketo Buildpacks 
https://github.com/paketo-community/go-generate/pull/88
https://github.com/paketo-buildpacks/php-web/pull/200

---

##### docker info
```
Client: Docker Engine - Community
 Version:           20.10.3
 API version:       1.41
 Go version:        go1.13.15
 Git commit:        48d30b5
 Built:             Fri Jan 29 14:33:13 2021
 OS/Arch:           linux/amd64
 Context:           default
 Experimental:      true

Server: Docker Engine - Community
 Engine:
  Version:          20.10.6
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.13.15
  Git commit:       8728dd2
  Built:            Fri Apr  9 22:44:13 2021
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.4.4
  GitCommit:        05f951a3781f4f2c1911b05e61c160e9c30eaa8e
 runc:
  Version:          1.0.0-rc93
  GitCommit:        12644e614e25b05da6fd08a38ffa0cfe1903fdec
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

**Comment by jromero** [comment]
Likely related to changes in https://github.com/buildpacks/pack/pull/1212

**Comment by samj1912** [comment]
This also seems to be impacting me. Any chance we could revert this behavior to what it was before?

**Comment by jromero** [comment]
Getting out a patch!

Resolved by e09e397bc45315fa537fded032e57e678525c0b4

**Comment by samj1912** [comment]
Thank you so much for the quick fix!
